staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/defaults.go

```diff
@@ -72,6 +73,38 @@ func SetDefaults_CustomResourceDefinitionSpec(obj *CustomResourceDefinitionSpec)
 	if obj.PreserveUnknownFields == nil {
 		obj.PreserveUnknownFields = utilpointer.BoolPtr(true)
 	}
+
+	// See if we need to default the schema.
+	// CRDs that specify spec.preserveUnknownFields=false are required to provide their own schemas.
+	if obj.Validation == nil && *obj.PreserveUnknownFields {
+		// Set up a minimal default
+		defaultSchema := &CustomResourceValidation{OpenAPIV3Schema: &JSONSchemaProps{Type: "object", XPreserveUnknownFields: utilpointer.BoolPtr(true)}}
```

Comment:
This will break v1beta1 CRD with non-structural schema
https://github.com/kubernetes/kubernetes/blob/f4e39afea082afac7bf6f6936c34e012b7cb3c25/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L180-L182

perhaps we should scope `mustBeStructural` to each version

Reply:
good point. Will add a test and stew on what to do here. cc @sttts

Reply:
we can require schema.openAPIV3Schema for create via v1, and require it for update via v1 if all versions in the existing object already have it

Reply:
will drop the defaulting for v1beta1

Reply:
dropped the defaulting for v1beta1, used ratcheting validation to require schemas in v1